lib/FreeRTOS/queue.c

```diff
@@ -751,7 +751,9 @@ TimeOut_t xTimeOut;
 Queue_t * const pxQueue = xQueue;
 
 	configASSERT( pxQueue );
-	configASSERT( !( ( pvItemToQueue == NULL ) && ( pxQueue->uxItemSize != ( UBaseType_t ) 0U ) ) );
+	/* A queue storage area should be provided if the item size is not 0, and
+	   should not be provided if the item size is 0. */
+	configASSERT( ( pvItemToQueue != NULL ) || ( pxQueue->uxItemSize == ( UBaseType_t ) 0U ) );
```

Comment:
I think it would be simpler to comment that the assertion says "If pvItemToQueue is null, then pxQueue->uxItemSize is 0".